src/eins-hwinfo.c

```diff
@@ -105,9 +115,21 @@ round_to_nearest (guint64 size,
   return (guint32) CLAMP (size / divisor, 0, G_MAXUINT32);
 }
 
-GVariant *
-eins_hwinfo_get_disk_space_for_partition (GFile   *file,
-                                          GError **error)
+guint32
+eins_hwinfo_get_ram_size (void)
+{
+  glibtop_mem mem;
+  guint32 size_mib;
+
+  glibtop_get_mem (&mem);
+  size_mib = round_to_nearest (mem.total, ONE_MIB_IN_BYTES);
+  return size_mib;
+}
+
+void
```

Comment:
Conventionally this function would return `TRUE` (with `diskspace` set and `error` untouched) or `FALSE` (with `diskspace` untouched and `error` set).

```suggestion
gboolean
```

and then adjust callers to use

```
if (eins_hwinfo_get_disk_space_for_partition (...))
  {
    // success path
  }
else
  {
    // error path
  }
```

The other option is to log the error within this function, initialise `diskspace` to (0, 0, 0), and remove the `error` parameter. This might be better since it would allow the rest of the event to be sent.

Reply:
I prefer the later one which simplifies things.